aea/aea.py

```diff
@@ -37,7 +39,7 @@ class AEA(Agent):
 
     def __init__(self, name: str,
                  mailbox: MailBox,
-                 private_key_pem_path: Optional[str] = None,
+                 wallet: Wallet,
```

Comment:
We had considered passing in the wallet before. Since the MailBox depends on the Wallet this makes sense.